nutils/function.py

```diff
@@ -2793,8 +2793,8 @@ class PlainBasis(Basis):
   __slots__ = '_coeffs', '_dofs'
 
   def __init__(self, coefficients: Sequence[numpy.ndarray], dofs: Sequence[numpy.ndarray], ndofs: int, index: Array, coords: Array) -> None:
-    self._coeffs = tuple(map(types.frozenarray, coefficients))
-    self._dofs = tuple(map(types.frozenarray, dofs))
+    self._coeffs = tuple(types.frozenarray(c, dtype=float) for c in coefficients)
+    self._dofs = tuple(types.frozenarray(d, dtype=int) for d in dofs)
```

Comment:
I don't see why this is necessary. Especially since the last commit reverts the cast.

Reply:
This is necessary in relation to checking, rather than casting, the dtype in evaluable.Elemwise. The last commit no longer requires this as `dataclass` automatically converts data to the canonical type for its kind.